openedx/core/djangoapps/programs/rest_api/v1/views.py

```diff
@@ -170,26 +173,22 @@ def transform_authoring_organizations(authoring_organizations) -> list[dict[str,
 
         return programs
 
+    @enterprise_is_enabled(otherwise=EmptyQuerySet)
     def _get_enterprise_course_enrollments(
         self, enterprise_uuid: str, user: "AnonymousUser | User"
-    ) -> list[CourseEnrollment]:
+    ) -> "QuerySet[CourseEnrollment]":
         """
         Return only enterprise enrollments for a user.
         """
-        enterprise_enrollment_course_ids = list(
-            EnterpriseCourseEnrollment.objects.filter(
-                enterprise_customer_user__user_id=user.id,
-                enterprise_customer_user__enterprise_customer__uuid=enterprise_uuid,
-            ).values_list("course_id", flat=True)
+        enterprise_enrollment_course_ids = (
+            get_enterprise_course_enrollments(user)
+            .filter(enterprise_customer_user__enterprise_customer__uuid=enterprise_uuid)
+            .values_list("course_id", flat=True)
         )
 
-        course_enrollments = (
-            CourseEnrollment.enrollments_for_user(user)
-            .filter(course_id__in=enterprise_enrollment_course_ids)
-            .select_related("course")
-        )
+        course_enrollments = get_course_enrollments(user, True, (enterprise_enrollment_course_ids))
 
-        return list(course_enrollments)
+        return course_enrollments
```

Comment:
I'm not a big fan of passing around lists, when you could pass around a QuerySet, so I converted this to return the QuerySet. If I had my druthers the `ProgramProgressMeter` would also take a QuerySet, but honestly at some point that whole thing is going to get re-factored in some way or another.